### PR TITLE
Removed resource_type attribute and replaced it with solr-only attribute human_readable_type.

### PR DESCRIPTION
**Pull request by val99erie**
This work is in preparation for issue ndlib/planning#247, but I wanted to submit it as its own pull request since it's a big logical chunk of work.

### CODE REVIEW

File: spec/models/collection_spec.rb [hunk 1]
```diff
@@ -25,19 +25,18 @@
   describe "to_solr on a saved object" do
     before {subject.save(validate: false)}
     let(:solr_doc) {subject.to_solr}
+
     describe "when profile is set" do
-      before { subject.resource_type = 'Profile' }
```
Comment (jcoyne) [inline spec/models/collection_spec.rb:29]: On a collection I think it is right to set resource_type to Collection vs Profile (and store that in descMetadata).  In the case of Collection, I think human_readable_type should be dependent on what resource_type is set to.

File: app/repository_models/collection.rb [hunk 2]
```diff
@@ -6,18 +6,13 @@ class Collection < ActiveFedora::Base
...
-  # Reads from resource_type attribute.
   # Defaults to "Collection", but can be set to something else.
-  # Profiles are marked with resource_type of "Profile" when they're created by the associated Person object
+  # Profiles are marked with human_readable_type of "Profile" when they're created by the associated Person object
   # This is used to populate the Object Type Facet
   def human_readable_type
-    self.resource_type ||= "Collection"
+    @human_readable_type ||= 'Collection'
   end
+  attr_writer :human_readable_type
```
Comment (jcoyne) [inline app/repository_models/collection.rb]: I don't think we need this (see my other comment below regarding Profile as a subtype of Collection).

**Comment by jeremyf** [comment]
@val99erie the build broke based on upstream changes to Hydra modules (as introduced by yours truly)

Once #183 is merged, we'll need to restart this job.

**Comment by val99erie** [comment]
@jeremyf  No problem, I have some changes to make after code review anyway, so no need to re-start the build just yet.

**Comment by jcoyne** [comment]
:+1:

**Pull request merged**